### PR TITLE
Added git commit validation message.

### PR DESCRIPTION
**Pull request by db-Matroskeen**

### CODE REVIEW

File: scripts/git_hooks/commit-msg [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+
+# Jira Issue is required. Other options (like #commenr) are optional, but appreciated.
```
Comment (bohart) [inline scripts/git_hooks/commit-msg]: like #commenr > like #comment ?

File: scripts/git_hooks/commit-msg [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+# More details: https://support.atlassian.com/jira-software-cloud/docs/process-issues-with-smart-commits
+
```
Comment (bohart) [inline scripts/git_hooks/commit-msg]:
```
# Helpers: color patterns.
RED='\033[0;31m'
NC='\033[0m'

error_msg ="\n ${RED}[ERROR] Aborting commit.${NC} Please add Jira Issue to your commit message. Example: 'OI-01: #comment Added cool feature.'\n\n"

```

File: scripts/git_hooks/commit-msg [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+commit_regex='(OI-[0-9]+)'
+error_msg="Aborting commit. Please add Jira Issue to your commit message. Example: 'OI-01: #comment Added cool feature.'"
```
Comment (bohart) [inline scripts/git_hooks/commit-msg]: Should end with new line sign. `\n\n`

File: scripts/install-git-hooks.sh [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+# Add git hooks to our both git repositories.
+cp scripts/git_hooks/* .git/hooks/
+cp scripts/git_hooks/* web/profiles/contrib/idea/.git/hooks/
+
+# Remove sample files.
+cd scripts/git_hooks
+for file in *; do
+  sample="$file.sample"
+  rm -f "$docroot/.git/hooks/$sample"
```
Comment (bohart) [inline scripts/install-git-hooks.sh]: Why do we need this for project development?

File: scripts/install-git-hooks.sh [hunk 5]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
```
Comment (bohart) [inline scripts/install-git-hooks.sh:1]: Here is my sample:

```
#!/usr/bin/env bash

add_hook () {
  WORKING_DIR=$PWD
  GIT_DIR="web/profiles/contrib/idea/.git/hooks"
  SCRIPT_DIR="../../scripts/git-hooks" 

  # Change our working directory.
  cd ${GIT_DIR}

  # Remove old hooks (the file and the link).
  [ -e ${1} ] && rm ${1}
  [ -L ${1} ] && rm ${1}

  # Ensure we have set all scripts as executable.
  cp "${SCRIPT_DIR}/${1}.sh" "./${1}"
  chmod 777 "./${1}"
  chmod +x "./${1}"

  # Change the working directory back.
  cd ${WORKING_DIR}
}

add_hook commit-msg
add_hook pre-commit
add_hook pre-push
```

File: composer.json [hunk 6]
```diff
@@ -30,7 +30,8 @@
     },
     "scripts": {
         "post-install-cmd": [
```
Comment (bohart) [inline composer.json:32]: should be added to post-update-cmd as well

File: composer.json [hunk 7]
```diff
@@ -30,7 +30,8 @@
...
-            "rm -rf web/libraries/dropzone/test"
+            "rm -rf web/libraries/dropzone/test",
```
Comment (bohart) [inline composer.json:33]: should be added to post-update-cmd as well

File: scripts/install-git-hooks.sh (hunk 5 above)
Comment (db-Matroskeen) [reply scripts/install-git-hooks.sh:1]: I've applied some changes based on this snippet, thanks!

**Pull request merged**